core/src/main/java/com/yahoo/oak/DeletedMemoryAccessException.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+public class DeletedMemoryAccessException extends RuntimeException {
```

Comment:
Why is it a runtime exception? Just to avoid declaring it?
Please change it to a regular exception.

Reply:
I think it is cleaner that way, since it is un-checked exception.
Any reason why to not use RuntimeException ?

Reply:
Here is a good explanation: https://stackoverflow.com/a/11589380/2570677
Since this error is not the programmer's fault and it is recoverable, then it should be checked.

Reply:
This turned up to be not such good idea, although it is more "right to do".

especially in the iterator, since iterator is not declared to throw stuff, we need to deal with this and to try and catch in the iterator code, which results in throwing a runtime exception at the end.

Moreover this exception is not meant to be for the programmer, it is internal to Oak, and we retry when get such exception.

Reply:
Your statements are contridicting. If you handle the exception internally, then you have to change the iterator code anyway.
This is all the more reason to change this exception to be checked. So we can validate that we never throw this exception to the user.
Right now, with your current implementation, this exception might be aggregated to the user without you noticing it.